pkg/frontend/v2/frontend_scheduler_adapter.go

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package v2
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/dskit/httpgrpc"
+	"github.com/grafana/dskit/tenant"
+
+	"github.com/grafana/mimir/pkg/frontend/querymiddleware"
+	"github.com/grafana/mimir/pkg/querier"
+	"github.com/grafana/mimir/pkg/scheduler/schedulerpb"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+type frontendToSchedulerAdapter struct {
+	cfg    Config
+	limits Limits
+}
+
+func (a *frontendToSchedulerAdapter) frontendToSchedulerEnqueueRequest(
+	req *frontendRequest, frontendAddr string,
+) (*schedulerpb.FrontendToScheduler, error) {
+	addlQueueDims, err := a.extractAdditionalQueueDimensions(req.ctx, req.request, time.Now())
```

Comment:
I think this should be behind a feature flag - so we can turn on the additional dimensions where we want to test them, but they're not on by default just yet.